pb/summary/summary.proto

```diff
@@ -45,6 +47,74 @@ message FailingTestSummary {
   string fail_test_link = 13;
 }
 
```

Comment:
I want to back up a little bit and look at the design of the additional proto stuff here before this gets checked in. (Proto fields are easy enough to deprecate and change later, but might be worth avoiding that). Sorry to do this midway through review! ><

Main two things are:
1. Some of these concepts (namely 'env') are GKE-specific and not generalized; I'm not super comfortable encoding that at the moment? If we do want to do that, I think it should be clearly marked that it's not general, and probably not integrated into summary.proto yet. (If it's a separate proto, it can get imported easily into here later).
2. Some of the status accumulation and whatnot might be similar to clusters in state.proto (https://github.com/GoogleCloudPlatform/testgrid/blob/master/pb/state/state.proto#L200)? Which might mean that's worth reusing. (I'm looking at infra_issues below for this).
3. Some of the info in TestInfo overlaps with FailingTestSummary; they serve different purposes (failing tests vs tests in general), but it'd be good if naming can stay consistent between them (display_name, linked_bugs).

@krzyzacy if you've also got thoughts

tl;dr If we're integrating this officially into Summarizer/summary.proto I want to be careful about making non-general changes, because they'll require cleaning-up later. If we're not, then this should be marked as a separate feature until we're ready to integrate it back.

Reply:
Piggy Backing a bit having just gone through the exercise of reverse engineering protos, avoiding duplication and being very specific about exactly what a field is via the comment is super valuable. We over load the term `name`, `target`, and `test group`. I think making the field name more descriptive is worth extra character length

Reply:
Done. I made changes to the naming/comments of fields to better match the existing field names that they should be related or the same as.